### PR TITLE
Remove all version checks for Java 6 and 7

### PR DESCRIPTION
**Pull request by chrisvest**
Motivation:
Netty 4.2 now requires Java 8 as the minimum supported version, so there is no need to guard for older JDKs.

Modification:
Remove all the version checks for Java 7 and older, and remove all uses of `@SuppressJava6Requirement`. Revapi complains these annotation removals are a potential compatibility issue, so ignores have been added.

Result:
Less code dealing with JDK versions.

### CODE REVIEW

File: codec-http/src/main/java/io/netty/handler/codec/spdy/SpdyHeaderBlockZlibEncoder.java [hunk 1]
```diff
@@ -73,17 +71,12 @@ private ByteBuf encode(ByteBufAllocator alloc, int len) {
...
         byte[] out = compressed.array();
         int off = compressed.arrayOffset() + compressed.writerIndex();
         int toWrite = compressed.writableBytes();
         final int numBytes;
-        if (PlatformDependent.javaVersion() >= 7) {
-            numBytes = compressor.deflate(out, off, toWrite, Deflater.SYNC_FLUSH);
-        } else {
-            numBytes = compressor.deflate(out, off, toWrite);
-        }
+        numBytes = compressor.deflate(out, off, toWrite, Deflater.SYNC_FLUSH);
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/spdy/SpdyHeaderBlockZlibEncoder.java:79]: You can merge the two lines

File: handler/src/main/java/io/netty/handler/ssl/Java8SslUtils.java [hunk 2]
```diff
@@ -75,7 +73,7 @@ static List getSniHostNames(List<String> names) {
         return sniServerNames;
     }
 
-    static List getSniHostName(byte[] hostname) {
+    static List<SNIServerName> getSniHostName(byte[] hostname) {
```
Comment (normanmaurer) [inline handler/src/main/java/io/netty/handler/ssl/Java8SslUtils.java:76]: We could just remove this class and merge it into its users ?

**Review by normanmaurer — CHANGES_REQUESTED** [review]

**Comment by chrisvest** [comment]
@normanmaurer Addressed comments and rebased.

**Review by normanmaurer — APPROVED** [review]
Ship it

**Comment by normanmaurer** [comment]
@chrisvest thanks!

**Pull request merged**